noir-projects/aztec-nr/aztec/src/oracle/aes128_decrypt.nr

```diff
@@ -109,12 +113,10 @@ mod test {
         let mut bad_sym_key = sym_key;
         bad_sym_key[0] = 0;
 
-        let received_plaintext = aes128_decrypt_oracle_wrapper(ciphertext, iv, bad_sym_key);
+        let received_plaintext = aes128_decrypt_oracle(ciphertext, iv, bad_sym_key);
 
-        let mut extracted_mac_as_bytes = [0 as u8; TEST_MAC_LENGTH];
-        for i in 0..TEST_MAC_LENGTH {
-            extracted_mac_as_bytes[i] = received_plaintext[TEST_PLAINTEXT_LENGTH + i];
-        }
+        let extracted_mac_as_bytes: [u8; TEST_MAC_LENGTH] =
+            subarray(received_plaintext.storage(), TEST_PLAINTEXT_LENGTH);
```

Comment:
Isn't this lovely?

Reply:
`subarray`? I think its constraints can be improved, but sure!

Reply:
Go for it!